app/controllers/katello/api/v2/activation_keys_controller.rb

```diff
@@ -347,7 +347,9 @@ def handle_blank_cve_params
 
     def single_assignment?
       (params.key?(:environment_id) && params.key?(:content_view_id)) ||
-      (params.key?(:environment) && params.key?(:content_view))
+      (params.key?(:environment) && params.key?(:content_view)) ||
+      (params.key?(:environment) && params.key?(:content_view_id)) || # AngularJS sends this on create
+      (params.key?(:environment_id) && params.key?(:content_view))
```

Comment:
What do you think about this to combine all the above conditions? Also, is it still a valid param combination if params has all 4 keys?
```
(params.key?(:environment) || params.key?(:environment_id)) && 
(params.key?(:content_view)) || params.key?(:content_view_id))
```

Reply:
I like that! will update.

And yes, that's handled by `find_cve_for_single`. I think it would ignore the `_id` params in that case and use `params.dig(:environment, :id)` and `params.dig(:content_view, :id)`.